lib/mustache_context.dart

```diff
@@ -124,12 +124,30 @@ class _MustacheContext extends MustacheToString implements MustacheContext {
   }
 
   dynamic _getActualValue(String key) {
-    try {
-      return ctx[key];
-    } catch (NoSuchMethodError) {
-      //Try to make dart2js understand that when we define USE_MIRRORS = false
-      //we do not want to use any reflector
-      return (useMirrors && USE_MIRRORS) ? ctxReflector[key] : null;
+    //Try to make dart2js understand that when we define USE_MIRRORS = false
+    //we do not want to use any reflector as that inflates the generated
+    //javascript.
+    if (useMirrors && USE_MIRRORS) {
+      if (reflect(ctx).type.instanceMembers.containsKey(new Symbol("[]"))) {
+        MethodMirror m = reflect(ctx).type.instanceMembers[new Symbol("[]")];
+        TypeMirror reflectedString = reflectType(String);
+        if (reflectedString.isAssignableTo(m.parameters[0].type)) {
```

Comment:
That exception wouldn't be trapped in any case, because UnimplementedError is not a subclass of NoSuchMethodError.  So it seems unlikely that reverting this change would correct that test.  Does it?